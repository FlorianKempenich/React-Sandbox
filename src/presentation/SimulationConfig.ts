export type ValueConfig = {
    min: number;
    max: number;
    default: number;
};
export type Config = {
    hourlyRate: ValueConfig;
    hoursPerDay: ValueConfig;
    daysPerMonth: ValueConfig;
    monthsPerYear: ValueConfig;
};
export const CONFIG: Config = {
    hourlyRate: {
        min: 20,
        max: 80,
        default: 30
    },
    hoursPerDay: {
        min: 2,
        max: 10,
        default: 6
    },
    daysPerMonth: {
        min: 5,
        max: 25,
        default: 10
    },
    monthsPerYear: {
        min: 2,
        max: 12,
        default: 6
    }
};