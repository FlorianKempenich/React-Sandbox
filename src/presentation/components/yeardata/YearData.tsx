import * as React from 'react';
import * as update from 'immutability-helper';
import ValueSlider from '../valueslider/ValueSlider';
import { style, captions } from './YearDataStyle';
import { ValueConfig } from '../../SimulationConfig';

export type YearDataProps = {
    onNewMoneyMadeInAYear: (moneyMade: number) => void;
    hourlyRateConfig: ValueConfig;
    hoursPerDayConfig: ValueConfig;
    daysPerMonthConfig: ValueConfig;
    monthsPerYearConfig: ValueConfig;
};
export interface YearDataState extends YearDataProps {
    hourlyRate: number;
    hoursPerDay: number;
    daysPerMonth: number;
    monthsPerYear: number;
}
type GrossAmountMadeParams = {
    hourlyRate?: number;
    hoursPerDay?: number;
    daysPerMonth?: number;
    monthsPerYear?: number;
};
export default class YearData extends React.Component<YearDataProps, YearDataState> {

    constructor(props: YearDataProps) {
        super(props);

        const initValues = {
            hourlyRate: props.hourlyRateConfig.default,
            hoursPerDay: props.hoursPerDayConfig.default,
            daysPerMonth: props.daysPerMonthConfig.default,
            monthsPerYear: props.monthsPerYearConfig.default
        };

        this.state = update(props, { $merge: initValues });

        this.handleNewHourlyRate = this.handleNewHourlyRate.bind(this);
        this.handleNewHoursPerDay = this.handleNewHoursPerDay.bind(this);
        this.handleNewDaysPerMonth = this.handleNewDaysPerMonth.bind(this);
        this.handleNewMonthsPerYear = this.handleNewMonthsPerYear.bind(this);
    }

    render() {
        // Paper
        return (
            <div style={style.root}>
                <div>
                    <p>10800</p>
                </div>
                <ValueSlider
                    mainCaption={captions.hourlyRate.mainCaption}
                    valueCaption={captions.hourlyRate.valueCaption}
                    min={this.state.hourlyRateConfig.min}
                    max={this.state.hourlyRateConfig.max}
                    onNewValue={this.handleNewHourlyRate}
                    value={this.state.hourlyRate}
                />
                <ValueSlider
                    mainCaption={captions.hoursPerDay.mainCaption}
                    valueCaption={captions.hoursPerDay.valueCaption}
                    min={this.state.hoursPerDayConfig.min}
                    max={this.state.hoursPerDayConfig.max}
                    onNewValue={this.handleNewHoursPerDay}
                    value={this.state.hoursPerDay}
                />
                <ValueSlider
                    mainCaption={captions.daysPerMonth.mainCaption}
                    valueCaption={captions.daysPerMonth.valueCaption}
                    min={this.state.daysPerMonthConfig.min}
                    max={this.state.daysPerMonthConfig.max}
                    onNewValue={this.handleNewDaysPerMonth}
                    value={this.state.daysPerMonth}
                />
                <ValueSlider
                    mainCaption={captions.monthsPerYear.mainCaption}
                    valueCaption={captions.monthsPerYear.valueCaption}
                    min={this.state.monthsPerYearConfig.min}
                    max={this.state.monthsPerYearConfig.max}
                    onNewValue={this.handleNewMonthsPerYear}
                    value={this.state.monthsPerYear}
                />
            </div>

        );
    }

    private handleNewHourlyRate(newHourlyRate: number) {
        console.log('Warning: For now doing nothing here, only setting state');
        this.setState(update(this.state, { $merge: { currentHourlyRate: newHourlyRate } }));
    }

    private handleNewHoursPerDay(newHoursPerDay: number) {
        console.log('Warning: For now doing nothing here, only setting state');
        this.setState(update(this.state, { $merge: { currentHoursPerDay: newHoursPerDay } }));
    }

    private handleNewDaysPerMonth(newDaysPerMonth: number) {
        console.log('Warning: For now doing nothing here, only setting state');
        this.setState(update(this.state, { $merge: { currentDaysPerMonth: newDaysPerMonth } }));
    }

    private handleNewMonthsPerYear(newMonthsPerYear: number) {
        console.log('Warning: For now doing nothing here, only setting state');
        this.setState(update(this.state, { $merge: { currentMonthsPerYear: newMonthsPerYear } }));
    }

    private calculateNewGrossAmountMade(newValues: GrossAmountMadeParams) {
        // TODO add test
    }

}