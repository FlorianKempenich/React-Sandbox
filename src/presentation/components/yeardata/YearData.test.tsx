import YearData from './YearData';
import * as React from 'react';
import { Config } from '../../SimulationConfig';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MuiThemeProvider } from 'material-ui/styles';

const MOCK_CONFIG: Config = {
    hourlyRate: {
        min: 20,
        max: 80,
        default: 30
    },
    hoursPerDay: {
        min: 2,
        max: 10,
        default: 6
    },
    daysPerMonth: {
        min: 5,
        max: 25,
        default: 10
    },
    monthsPerYear: {
        min: 2,
        max: 12,
        default: 6
    }
};

it('should render correctly', () => {
    const moneyMadeCallback = jest.fn();
    const yearDataJsx = (
        <MuiThemeProvider>
            <YearData
                onNewMoneyMadeInAYear={moneyMadeCallback}
                hourlyRate={MOCK_CONFIG.hourlyRate}
                hoursPerDay={MOCK_CONFIG.hoursPerDay}
                daysPerMonth={MOCK_CONFIG.daysPerMonth}
                monthsPerYear={MOCK_CONFIG.monthsPerYear}
            />
        </MuiThemeProvider>
    );
    const wrapper = mount(yearDataJsx);

    expect(toJson(wrapper)).toMatchSnapshot();
});
