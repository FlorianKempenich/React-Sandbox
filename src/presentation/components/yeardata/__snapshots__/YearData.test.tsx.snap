// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<MuiThemeProvider>
  <YearData
    daysPerMonth={
      Object {
        "default": 10,
        "max": 25,
        "min": 5,
      }
    }
    hourlyRate={
      Object {
        "default": 30,
        "max": 80,
        "min": 20,
      }
    }
    hoursPerDay={
      Object {
        "default": 6,
        "max": 10,
        "min": 2,
      }
    }
    monthsPerYear={
      Object {
        "default": 6,
        "max": 12,
        "min": 2,
      }
    }
    onNewMoneyMadeInAYear={[Function]}
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-around",
          "margin": 5,
          "width": 150,
        }
      }
    >
      <ValueSlider
        mainCaption="h. rate"
        max={80}
        min={20}
        onNewValue={[Function]}
        value={30}
        valueCaption="€ / hour"
      >
        <div
          className="value-slider"
        >
          <div
            className="main-caption"
          >
            <div>
              <p
                key="0"
              >
                h
              </p>
              <p
                key="1"
              >
                .
              </p>
              <p
                key="2"
              >
                 
              </p>
              <p
                key="3"
              >
                r
              </p>
              <p
                key="4"
              >
                a
              </p>
              <p
                key="5"
              >
                t
              </p>
              <p
                key="6"
              >
                e
              </p>
            </div>
          </div>
          <div
            className="value"
          >
            <NumberInput
              onNewValue={[Function]}
              value={30}
            >
              <input
                id="debug"
                inputMode="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value="30"
              />
            </NumberInput>
          </div>
          <div
            className="value-caption"
          >
            <p>
              € / hour
            </p>
          </div>
          <div
            className="slider"
          >
            <Slider
              axis="y"
              disableFocusRipple={false}
              disabled={false}
              max={80}
              min={20}
              onChange={[Function]}
              required={true}
              sliderStyle={
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
              step={0.01}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              value={30}
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "width": "100%",
                  }
                }
              >
                <div
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "default",
                      "height": "100%",
                      "marginBottom": 10,
                      "marginTop": 10,
                      "muiPrepared": true,
                      "position": "relative",
                      "touchCallout": "none",
                      "userSelect": "none",
                      "width": 18,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "bottom": 0,
                        "height": "100%",
                        "left": 8,
                        "muiPrepared": true,
                        "position": "absolute",
                        "width": 2,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#00bcd4",
                          "bottom": 0,
                          "directionInvariant": true,
                          "height": "calc(16.666666666666664%)",
                          "marginTop": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#bdbdbd",
                          "directionInvariant": true,
                          "height": "calc(83.33333333333334%)",
                          "marginBottom": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "top": 0,
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "backgroundClip": "padding-box",
                          "backgroundColor": "#00bcd4",
                          "border": "0px solid transparent",
                          "borderRadius": "50%",
                          "bottom": "16.666666666666664%",
                          "boxSizing": "border-box",
                          "cursor": "pointer",
                          "directionInvariant": true,
                          "height": 12,
                          "left": 0,
                          "margin": "0 0 0 1px",
                          "muiPrepared": true,
                          "outline": "none",
                          "overflow": "visible",
                          "pointerEvents": "inherit",
                          "position": "absolute",
                          "transform": "translate(-50%, 50%)",
                          "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": 12,
                          "zIndex": 1,
                        }
                      }
                      tabIndex={0}
                    >
                      <FocusRipple
                        color="#00bcd4"
                        innerStyle={
                          Object {
                            "height": "300%",
                            "left": -12,
                            "top": -12,
                            "width": "300%",
                          }
                        }
                        show={false}
                        style={
                          Object {
                            "height": 12,
                            "overflow": "visible",
                            "width": 12,
                          }
                        }
                      >
                        <ScaleIn
                          enterDelay={0}
                          maxScale={0.85}
                          style={
                            Object {
                              "height": 12,
                              "left": 0,
                              "overflow": "visible",
                              "position": "absolute",
                              "top": 0,
                              "width": 12,
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component="div"
                            style={
                              Object {
                                "height": 12,
                                "left": 0,
                                "muiPrepared": true,
                                "overflow": "visible",
                                "position": "absolute",
                                "top": 0,
                                "width": 12,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "height": 12,
                                  "left": 0,
                                  "muiPrepared": true,
                                  "overflow": "visible",
                                  "position": "absolute",
                                  "top": 0,
                                  "width": 12,
                                }
                              }
                            />
                          </TransitionGroup>
                        </ScaleIn>
                      </FocusRipple>
                    </div>
                  </div>
                </div>
                <input
                  max={80}
                  min={20}
                  required={true}
                  step={0.01}
                  type="hidden"
                  value={30}
                />
              </div>
            </Slider>
          </div>
        </div>
      </ValueSlider>
      <ValueSlider
        mainCaption="h. day"
        max={10}
        min={2}
        onNewValue={[Function]}
        value={6}
        valueCaption="h / day"
      >
        <div
          className="value-slider"
        >
          <div
            className="main-caption"
          >
            <div>
              <p
                key="0"
              >
                h
              </p>
              <p
                key="1"
              >
                .
              </p>
              <p
                key="2"
              >
                 
              </p>
              <p
                key="3"
              >
                d
              </p>
              <p
                key="4"
              >
                a
              </p>
              <p
                key="5"
              >
                y
              </p>
            </div>
          </div>
          <div
            className="value"
          >
            <NumberInput
              onNewValue={[Function]}
              value={6}
            >
              <input
                id="debug"
                inputMode="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value="6"
              />
            </NumberInput>
          </div>
          <div
            className="value-caption"
          >
            <p>
              h / day
            </p>
          </div>
          <div
            className="slider"
          >
            <Slider
              axis="y"
              disableFocusRipple={false}
              disabled={false}
              max={10}
              min={2}
              onChange={[Function]}
              required={true}
              sliderStyle={
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
              step={0.01}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              value={6}
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "width": "100%",
                  }
                }
              >
                <div
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "default",
                      "height": "100%",
                      "marginBottom": 10,
                      "marginTop": 10,
                      "muiPrepared": true,
                      "position": "relative",
                      "touchCallout": "none",
                      "userSelect": "none",
                      "width": 18,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "bottom": 0,
                        "height": "100%",
                        "left": 8,
                        "muiPrepared": true,
                        "position": "absolute",
                        "width": 2,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#00bcd4",
                          "bottom": 0,
                          "directionInvariant": true,
                          "height": "calc(50%)",
                          "marginTop": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#bdbdbd",
                          "directionInvariant": true,
                          "height": "calc(50%)",
                          "marginBottom": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "top": 0,
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "backgroundClip": "padding-box",
                          "backgroundColor": "#00bcd4",
                          "border": "0px solid transparent",
                          "borderRadius": "50%",
                          "bottom": "50%",
                          "boxSizing": "border-box",
                          "cursor": "pointer",
                          "directionInvariant": true,
                          "height": 12,
                          "left": 0,
                          "margin": "0 0 0 1px",
                          "muiPrepared": true,
                          "outline": "none",
                          "overflow": "visible",
                          "pointerEvents": "inherit",
                          "position": "absolute",
                          "transform": "translate(-50%, 50%)",
                          "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": 12,
                          "zIndex": 1,
                        }
                      }
                      tabIndex={0}
                    >
                      <FocusRipple
                        color="#00bcd4"
                        innerStyle={
                          Object {
                            "height": "300%",
                            "left": -12,
                            "top": -12,
                            "width": "300%",
                          }
                        }
                        show={false}
                        style={
                          Object {
                            "height": 12,
                            "overflow": "visible",
                            "width": 12,
                          }
                        }
                      >
                        <ScaleIn
                          enterDelay={0}
                          maxScale={0.85}
                          style={
                            Object {
                              "height": 12,
                              "left": 0,
                              "overflow": "visible",
                              "position": "absolute",
                              "top": 0,
                              "width": 12,
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component="div"
                            style={
                              Object {
                                "height": 12,
                                "left": 0,
                                "muiPrepared": true,
                                "overflow": "visible",
                                "position": "absolute",
                                "top": 0,
                                "width": 12,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "height": 12,
                                  "left": 0,
                                  "muiPrepared": true,
                                  "overflow": "visible",
                                  "position": "absolute",
                                  "top": 0,
                                  "width": 12,
                                }
                              }
                            />
                          </TransitionGroup>
                        </ScaleIn>
                      </FocusRipple>
                    </div>
                  </div>
                </div>
                <input
                  max={10}
                  min={2}
                  required={true}
                  step={0.01}
                  type="hidden"
                  value={6}
                />
              </div>
            </Slider>
          </div>
        </div>
      </ValueSlider>
      <ValueSlider
        mainCaption="d. month"
        max={25}
        min={5}
        onNewValue={[Function]}
        value={10}
        valueCaption="d / month"
      >
        <div
          className="value-slider"
        >
          <div
            className="main-caption"
          >
            <div>
              <p
                key="0"
              >
                d
              </p>
              <p
                key="1"
              >
                .
              </p>
              <p
                key="2"
              >
                 
              </p>
              <p
                key="3"
              >
                m
              </p>
              <p
                key="4"
              >
                o
              </p>
              <p
                key="5"
              >
                n
              </p>
              <p
                key="6"
              >
                t
              </p>
              <p
                key="7"
              >
                h
              </p>
            </div>
          </div>
          <div
            className="value"
          >
            <NumberInput
              onNewValue={[Function]}
              value={10}
            >
              <input
                id="debug"
                inputMode="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value="10"
              />
            </NumberInput>
          </div>
          <div
            className="value-caption"
          >
            <p>
              d / month
            </p>
          </div>
          <div
            className="slider"
          >
            <Slider
              axis="y"
              disableFocusRipple={false}
              disabled={false}
              max={25}
              min={5}
              onChange={[Function]}
              required={true}
              sliderStyle={
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
              step={0.01}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              value={10}
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "width": "100%",
                  }
                }
              >
                <div
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "default",
                      "height": "100%",
                      "marginBottom": 10,
                      "marginTop": 10,
                      "muiPrepared": true,
                      "position": "relative",
                      "touchCallout": "none",
                      "userSelect": "none",
                      "width": 18,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "bottom": 0,
                        "height": "100%",
                        "left": 8,
                        "muiPrepared": true,
                        "position": "absolute",
                        "width": 2,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#00bcd4",
                          "bottom": 0,
                          "directionInvariant": true,
                          "height": "calc(25%)",
                          "marginTop": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#bdbdbd",
                          "directionInvariant": true,
                          "height": "calc(75%)",
                          "marginBottom": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "top": 0,
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "backgroundClip": "padding-box",
                          "backgroundColor": "#00bcd4",
                          "border": "0px solid transparent",
                          "borderRadius": "50%",
                          "bottom": "25%",
                          "boxSizing": "border-box",
                          "cursor": "pointer",
                          "directionInvariant": true,
                          "height": 12,
                          "left": 0,
                          "margin": "0 0 0 1px",
                          "muiPrepared": true,
                          "outline": "none",
                          "overflow": "visible",
                          "pointerEvents": "inherit",
                          "position": "absolute",
                          "transform": "translate(-50%, 50%)",
                          "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": 12,
                          "zIndex": 1,
                        }
                      }
                      tabIndex={0}
                    >
                      <FocusRipple
                        color="#00bcd4"
                        innerStyle={
                          Object {
                            "height": "300%",
                            "left": -12,
                            "top": -12,
                            "width": "300%",
                          }
                        }
                        show={false}
                        style={
                          Object {
                            "height": 12,
                            "overflow": "visible",
                            "width": 12,
                          }
                        }
                      >
                        <ScaleIn
                          enterDelay={0}
                          maxScale={0.85}
                          style={
                            Object {
                              "height": 12,
                              "left": 0,
                              "overflow": "visible",
                              "position": "absolute",
                              "top": 0,
                              "width": 12,
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component="div"
                            style={
                              Object {
                                "height": 12,
                                "left": 0,
                                "muiPrepared": true,
                                "overflow": "visible",
                                "position": "absolute",
                                "top": 0,
                                "width": 12,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "height": 12,
                                  "left": 0,
                                  "muiPrepared": true,
                                  "overflow": "visible",
                                  "position": "absolute",
                                  "top": 0,
                                  "width": 12,
                                }
                              }
                            />
                          </TransitionGroup>
                        </ScaleIn>
                      </FocusRipple>
                    </div>
                  </div>
                </div>
                <input
                  max={25}
                  min={5}
                  required={true}
                  step={0.01}
                  type="hidden"
                  value={10}
                />
              </div>
            </Slider>
          </div>
        </div>
      </ValueSlider>
      <ValueSlider
        mainCaption="months"
        max={12}
        min={2}
        onNewValue={[Function]}
        value={6}
        valueCaption="m worked"
      >
        <div
          className="value-slider"
        >
          <div
            className="main-caption"
          >
            <div>
              <p
                key="0"
              >
                m
              </p>
              <p
                key="1"
              >
                o
              </p>
              <p
                key="2"
              >
                n
              </p>
              <p
                key="3"
              >
                t
              </p>
              <p
                key="4"
              >
                h
              </p>
              <p
                key="5"
              >
                s
              </p>
            </div>
          </div>
          <div
            className="value"
          >
            <NumberInput
              onNewValue={[Function]}
              value={6}
            >
              <input
                id="debug"
                inputMode="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value="6"
              />
            </NumberInput>
          </div>
          <div
            className="value-caption"
          >
            <p>
              m worked
            </p>
          </div>
          <div
            className="slider"
          >
            <Slider
              axis="y"
              disableFocusRipple={false}
              disabled={false}
              max={12}
              min={2}
              onChange={[Function]}
              required={true}
              sliderStyle={
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
              step={0.01}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              value={6}
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "width": "100%",
                  }
                }
              >
                <div
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "default",
                      "height": "100%",
                      "marginBottom": 10,
                      "marginTop": 10,
                      "muiPrepared": true,
                      "position": "relative",
                      "touchCallout": "none",
                      "userSelect": "none",
                      "width": 18,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "bottom": 0,
                        "height": "100%",
                        "left": 8,
                        "muiPrepared": true,
                        "position": "absolute",
                        "width": 2,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#00bcd4",
                          "bottom": 0,
                          "directionInvariant": true,
                          "height": "calc(40%)",
                          "marginTop": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "[object Object]": 0,
                          "backgroundColor": "#bdbdbd",
                          "directionInvariant": true,
                          "height": "calc(60%)",
                          "marginBottom": 6,
                          "muiPrepared": true,
                          "position": "absolute",
                          "top": 0,
                          "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "backgroundClip": "padding-box",
                          "backgroundColor": "#00bcd4",
                          "border": "0px solid transparent",
                          "borderRadius": "50%",
                          "bottom": "40%",
                          "boxSizing": "border-box",
                          "cursor": "pointer",
                          "directionInvariant": true,
                          "height": 12,
                          "left": 0,
                          "margin": "0 0 0 1px",
                          "muiPrepared": true,
                          "outline": "none",
                          "overflow": "visible",
                          "pointerEvents": "inherit",
                          "position": "absolute",
                          "transform": "translate(-50%, 50%)",
                          "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": 12,
                          "zIndex": 1,
                        }
                      }
                      tabIndex={0}
                    >
                      <FocusRipple
                        color="#00bcd4"
                        innerStyle={
                          Object {
                            "height": "300%",
                            "left": -12,
                            "top": -12,
                            "width": "300%",
                          }
                        }
                        show={false}
                        style={
                          Object {
                            "height": 12,
                            "overflow": "visible",
                            "width": 12,
                          }
                        }
                      >
                        <ScaleIn
                          enterDelay={0}
                          maxScale={0.85}
                          style={
                            Object {
                              "height": 12,
                              "left": 0,
                              "overflow": "visible",
                              "position": "absolute",
                              "top": 0,
                              "width": 12,
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component="div"
                            style={
                              Object {
                                "height": 12,
                                "left": 0,
                                "muiPrepared": true,
                                "overflow": "visible",
                                "position": "absolute",
                                "top": 0,
                                "width": 12,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "height": 12,
                                  "left": 0,
                                  "muiPrepared": true,
                                  "overflow": "visible",
                                  "position": "absolute",
                                  "top": 0,
                                  "width": 12,
                                }
                              }
                            />
                          </TransitionGroup>
                        </ScaleIn>
                      </FocusRipple>
                    </div>
                  </div>
                </div>
                <input
                  max={12}
                  min={2}
                  required={true}
                  step={0.01}
                  type="hidden"
                  value={6}
                />
              </div>
            </Slider>
          </div>
        </div>
      </ValueSlider>
    </div>
  </YearData>
</MuiThemeProvider>
`;
